08-interop/08-interop.md

```diff
@@ -169,3 +169,109 @@ Task #10 - UDP between lwIP and emb6 on iotlab-m3
+=====================================================
+### Description
+
+UDP packet exchange between an iotlab-m3 node running RIOT and a samr21-xpro
+node running Zephyr.
+
+### Testing procedure
+
+#### Setting up the Zephyr node
+
+1. Follow the
+['Getting Started'](https://docs.zephyrproject.org/latest/getting_started/index.html)
+guide on Zephyr's documentation.
+
+2. You will be using the `echo_server` example application. To compile and
+   flash, connect the samr21-xpro board and in zephyr's root directory run:
+
+   ```sh
+   west build -p auto -b atsamr21_xpro samples/net/sockets/echo_server -- -DOVERLAY_CONFIG=overlay-802154.conf
+   west flash
+   ```
+
+3. If everything is OK you should be able to connect to the serial output use
+   any terminal program. For example:
+
+   ```sh
+   minicom -D /dev/ttyACM0 -o
+   ```
+
+4. By pressing tab you should be able to see all available commands. Get the
+   node's IPv6 by running the following on its shell:
+
+   ```sh
+   # On samr21-xpro:zephyr node
+   uart:~$ net ipv6
+
+   IPv6 support                              : enabled
+   IPv6 fragmentation support                : disabled
+   Multicast Listener Discovery support      : enabled
+   Neighbor cache support                    : enabled
+   Neighbor discovery support                : enabled
+   Duplicate address detection (DAD) support : enabled
+   Router advertisement RDNSS option support : enabled
+   6lo header compression support            : enabled
+   Max number of IPv6 network interfaces in the system          : 1
+   Max number of unicast IPv6 addresses per network interface   : 3
+   Max number of multicast IPv6 addresses per network interface : 4
+   Max number of IPv6 prefixes per network interface            : 2
+
+   IPv6 addresses for interface 0x20007d60 (IEEE 802.15.4)
+   =======================================================
+   Type            State           Lifetime (sec)  Address
+   autoconf        preferred       infinite        fe80::d419:100:7ae4:9b3b/128
+   manual          preferred       infinite        2001:db8::1/128
+   ```
+
+   Similarly, to get the IEEE 802.15.4 PAN ID and channel, run:
+
+   ```sh
+   # On samr21-xpro:zephyr node
+   uart:~$ ieee802154 get_pan_id
+   PAN ID 43981 (0xabcd)
+
+   uart:~$ ieee802154 get_chan
+   Channel 26
+   ```
+
+
+#### Setting up the RIOT node
+
+1. Flash on the iotlab-m3 board the [gnrc_networking] example or the [gnrc_udp]
+   test.
+
+2. Set the channel and PAN ID to the same values as the Zephyr node, running:
+
+   ```sh
+   # On iotlab-m3:riot node
+   > ifconfig 6 set chan 26
+   ifconfig 6 set chan 26
+   success: set channel on interface 6 to 26
+
+   > ifconfig 6 set pan_id 0xabcd
+   ifconfig 6 set pan_id 0xabcd
+   success: set network identifier on interface 6 to 0xabcd
+   ```
+
+3. The UDP server application on the Zephyr node will be listening to port 4242.
+   Start the UDP server on the same port on the RIOT node:
```

Comment:
Does the echo server reply to that port?

Reply:
Yes, it echoes to the same port it is listening on.

Reply:
That's weird... why not to the source port?

Reply:
Sorry, I confused myself with another example I was reading. I checked [the code](https://github.com/zephyrproject-rtos/zephyr/blob/ba0aea3cd3f4aee1f21d3e1b798e6713827465f3/samples/net/sockets/echo_server/src/udp.c#L118-L119) and it sends back to the address as is, so it should be the source port.

Reply:
Ah, but we use the same source port as the destination port for `udp send`, right! Still, something like this should be mentioned here in my opinion to avoid confusion.

Reply:
Changed to clarify this point.